server/src/main/java/org/apache/druid/discovery/DiscoveryDruidNode.java

```diff
@@ -164,6 +166,13 @@ public DruidNode getDruidNode()
     return druidNode;
   }
 
+  @Nullable
+  @JsonIgnore
+  public <T extends DruidService> T getService(String key)
+  {
+    return (T) services.get(key);
+  }
+
```

Comment:
What are the chances that there's a ClassCastException thrown here?  Should we try to be defensive?

Reply:
like if rogue nodes announce things at the wrong key?

Reply:
went ahead and changed to do safer cast which changed the signature to need a `Class<T>` which seems ok :shrug: